### PR TITLE
Onboard users

### PR DESCRIPTION
**Pull request by maitra100**

### CODE REVIEW

File: src/controllers/addUser.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+const addUserController = async (req, res) => {
+  try {
+    await addUserService(req.body)
+    res.status(200).send('User added successfully')
+  } catch (e) {
+    res.status(500).send(e.message)
+  }
+}
+
+module.exports = addUserController
```
Comment (rkdixit96) [inline src/controllers/addUser.js:12]: rename to just user.js

File: src/services/addUser.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    firstName: user.firstName,
+    middleName: user.middleName,
+    lastName: user.lastName,
+    profilePhoto: user.profilePhoto,
+    emailId: user.emailId,
+    phoneNumber: user.phoneNumber,
+    address: user.address,
+    gender: user.gender,
+    dateOfBirth: user.dateOfBirth
+  }, { returning: true })
```
Comment (rkdixit96) [inline src/services/addUser.js:17]: Just spread the userObject inside create 
```
db.Users.create({
    ...user
}
```

File: database/migrations/20230304175042-changeGender.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  },
+
+  async down (queryInterface, Sequelize) {
+    return queryInterface.sequelize.transaction((t) => {
+      return Promise.all([
+        queryInterface.renameColumn('Users', 'gender', 'genders', { transaction: t })
+      ])
+    })
+  }
+}
```
Comment (rkdixit96) [inline database/migrations/20230304175042-changeGender.js:20]: Combine all these migrations into one migration

**Review by rkdixit96 — CHANGES_REQUESTED** [review]

File: src/controllers/addUser.js (hunk 1 above)
Comment (maitra100) [reply src/controllers/addUser.js]: done

File: src/services/addUser.js (hunk 2 above)
Comment (maitra100) [reply src/services/addUser.js]: done

File: database/migrations/20230304175042-changeGender.js (hunk 3 above)
Comment (maitra100) [reply database/migrations/20230304175042-changeGender.js]: done

File: src/util/middleware/userOnboardingValidator.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  lastName: Joi.string().max(100),
+  profilePhoto: Joi.string().max(10000).required(),
+  emailId: Joi.string().email().max(256).required(),
+  phoneNumber: Joi.string().min(10).max(10).required(),
+  address: Joi.string().required(),
+  gender: Joi.string().valid('M', 'F', 'O').required(),
+  dateOfBirth: Joi.string().required()
+})
+
+const userValidator = (req, res, next) => {
```
Comment (rkdixit96) [inline src/util/middleware/userOnboardingValidator.js:17]: Create generic validator that accepts schema as input

File: src/routes/user.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const router1 = require('express').Router()
+const addUserController = require('../controllers/user')
+const userValidator = require('../util/middleware/userOnboardingValidator')
+
+router1.post('/', userValidator, addUserController)
+
+module.exports = router1
```
Comment (rkdixit96) [inline src/routes/user.js:7]: Please don't use variables like `router1`. Just call them router

File: src/database/models/users.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    lastName: DataTypes.STRING,
+    profilePhoto: DataTypes.STRING,
+    emailId: DataTypes.STRING,
+    phoneNumber: DataTypes.STRING,
+    address: DataTypes.STRING,
+    dateOfBirth: DataTypes.DATEONLY,
+    gender: DataTypes.ENUM('M', 'F', 'O')
+  }, {
+    sequelize,
+    modelName: 'Users'
```
Comment (rkdixit96) [inline src/database/models/users.js:30]: Tables - use all small letters

**Review by rkdixit96 — CHANGES_REQUESTED** [review]

File: src/routes/user.js (hunk 5 above)
Comment (maitra100) [reply src/routes/user.js:7]: done

File: src/database/models/users.js (hunk 6 above)
Comment (maitra100) [reply src/database/models/users.js:30]: done

File: src/util/middleware/userOnboardingValidator.js (hunk 4 above)
Comment (maitra100) [reply src/util/middleware/userOnboardingValidator.js]: done

**Review by rkdixit96 — APPROVED** [review]

**Review by SaatwikRishi — APPROVED** [review]

**Pull request merged**